Lib/_colorize.py

```diff
@@ -86,6 +75,165 @@ class ANSIColors:
         setattr(NoColors, attr, "")
 
 
+#
+# Experimental theming support (see gh-133346)
+#
+
+# - Create a theme by copying an existing `Theme` with one or more sections
```

Comment:
If this is all the docs we are going to offer for now, let's add a full example here that can be copy pasted with the data class and maybe with the default_theme.copy_with().

Reply:
Agreed.